src/test/java/com/ordestiny/tdd/designpattern/factorypattern/AnimalFactoryTest.java

```diff
@@ -0,0 +1,77 @@
+package com.ordestiny.tdd.designpattern.factorypattern;
+
+import com.ordestiny.tdd.designpattern.factorypattern.animal.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * @author Rachit Parikh
+ */
+
+public class AnimalFactoryTest {
+
+    Animal animal;
+
+    @Test
+    public void checkDog() throws Exception{
+        animal = AnimalFactory.getAnimal(AnimalEnum.DOG);
+        assertEquals(animal.getClass(), Dog.class);
+        assertEquals(animal.numberOfLegs(), 4);
+        assertEquals(animal.typeOfAnimal(), AnimalType.MAMMAL);
+    }
+
+    @Test
+    public void checkHuman() throws Exception{
+        animal = AnimalFactory.getAnimal(AnimalEnum.HUMAN);
+        assertEquals(animal.getClass(), Human.class);
+        assertEquals(animal.numberOfLegs(), 2);
+        assertEquals(animal.typeOfAnimal(), AnimalType.MAMMAL);
+    }
+
+    @Test
+    public void checkSalmon() throws Exception{
+        animal = AnimalFactory.getAnimal(AnimalEnum.SALMON);
+        assertEquals(animal.getClass(), Salmon.class);
+        assertEquals(animal.numberOfLegs(), 0);
+        assertEquals(animal.typeOfAnimal(), AnimalType.FISH);
+    }
+
+    @Test
+    public void checkSnake() throws Exception{
+        animal = AnimalFactory.getAnimal(AnimalEnum.SNAKE);
+        assertEquals(animal.getClass(), Snake.class);
+        assertEquals(animal.numberOfLegs(), 0);
+        assertEquals(animal.typeOfAnimal(), AnimalType.REPTILE);
+    }
+
+    @Test
+    public void checkEarthworm() throws Exception{
+        animal = AnimalFactory.getAnimal(AnimalEnum.EARTHWORM);
+        assertEquals(animal.getClass(), Earthworm.class);
+        assertEquals(animal.numberOfLegs(), 0);
+        assertEquals(animal.typeOfAnimal(), AnimalType.INVERTEBRATE);
+    }
+
+    @Test
+    public void checkEagle() throws Exception{
+        animal = AnimalFactory.getAnimal(AnimalEnum.EAGLE);
+        assertEquals(animal.getClass(), Eagle.class);
+        assertEquals(animal.numberOfLegs(), 2);
+        assertEquals(animal.typeOfAnimal(), AnimalType.BIRD);
+    }
+
+    @Test
+    public void checkFrog() throws Exception{
+        animal = AnimalFactory.getAnimal(AnimalEnum.FROG);
+        assertEquals(animal.getClass(), Frog.class);
+        assertEquals(animal.numberOfLegs(), 2);
+        assertEquals(animal.typeOfAnimal(), AnimalType.AMPHIBIAN);
+    }
+
+    @Test
+    public void checkAnimalFactory(){

```

Comment:
What is the purpose of this test case?

Reply:
Initially I wrote it to check whether the object was being created or not, but I realized that the above test  cases are passing so this test was not required. Will remove it.